src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -197,17 +197,37 @@
   end
 
   describe 'GET #subprojects' do
+    let(:apache2) { create(:project, name: 'Apache:Apache2') }
+    let(:apache2_subproject1) { create(:project, name: 'Apache:Apache2:TestSubproject') }
+    let(:apache2_subproject2) { create(:project, name: 'Apache:Apache2:TestSubproject2') }
+    let(:apache2_subprojects) {[apache2_subproject1, apache2_subproject2]}
+
     before do
       apache_project
-      @project = create(:project, name: 'Apache:Apache2')
-      create(:project, name: 'Apache:Apache2:TestSubproject')
-      create(:project, name: 'Apache:Apache2:TestSubproject2')
-      another_project
-      get :subprojects, params: { project: @project }
+      apache2
+      apache2_subproject1
+      apache2_subproject2
```

Comment:
Instead of touching all the projects in the before you could just use the bang ``!`` operator on the let.
```
let!(:apache2_subproject1) { create(:project, name: 'Apache:Apache2:TestSubproject') }
```
should do the same as a normal let and touching apache2_subproject1 in the before.